lib/java/com/google/android/material/imageview/ShapeableImageView.java

```diff
@@ -159,7 +159,8 @@ private void drawStroke(Canvas canvas) {
   @Override
   public void setShapeAppearanceModel(@NonNull ShapeAppearanceModel shapeAppearanceModel) {
     this.shapeAppearanceModel = shapeAppearanceModel;
-    requestLayout();
+    /*Refreshing this View to apply the newest shapeAppearanceModel */
+    this.onSizeChanged(getWidth(),getHeight(),getWidth(),getHeight());
```

Comment:
Thanks for the PR!

You still need to call requestLayout to redraw. You can change it to invalidate now.
onSizeChanged won't force a redraw.  Please do the following:

- Remove comment
- Extract logic from onSizeChanged to an `updateShapeAppearance(int width, int height)` method. 
- Call that method here before invalidate and on `onSizeChanged`.
- Watch the formatting (spaces after commas)

Reply:
Hi @ymarian, I apologize if I misunderstood your comments. I moved the "refresh" part of `setShapeAppearanceModel` to a new `updateShapeAppearanceModel` and ran `updateShape...` from `setShape...`. I agree that the update logic should be decoupled from the setter. Thanks for the review 😉

Reply:
Oh yeah I was saying something different, no worries. I'll make the changes and submit this. Thanks for finding the bug!

Reply:
Cool, I just needed this bug fixed for an upcoming Lists PR...

Reply:
I am going to leave this PR open just in case I encounter this issue again after you submit the patch. My dev environment is very specific.